metaflow/plugins/kfp/kfp.py

```diff
@@ -557,14 +556,10 @@ def _set_container_resources(
         resource_requirements: Dict[str, Any] = kfp_component.resource_requirements
         if "memory" in resource_requirements:
             container_op.container.set_memory_request(resource_requirements["memory"])
-        if "memory_limit" in resource_requirements:
-            container_op.container.set_memory_limit(
-                resource_requirements["memory_limit"]
-            )
+            container_op.container.set_memory_limit(resource_requirements["memory"])
```

Comment:
do we really need to call `set_memory_limit`, doesn't it become equal to the request?

Reply:
By `doesn't it become equal to the request` are you referring to DevEx's work to enable this? If so,
(1) We're still not 100% sure of DevEx's design (whether they'll set limit to the request or just prevent the pod from being scheduled)
(2) Even if DevEx sets the limit to the request for us, they wanted this SDK change before they rolled out their change to improve the health of the cluster.

Reply:
Ok, true, we must set it, as outlined here:
> Every Container in the Pod must have a memory limit and a memory request.
https://kubernetes.io/docs/tasks/configure-pod-container/quality-service-pod/#create-a-pod-that-gets-assigned-a-qos-class-of-guaranteed